### PR TITLE
Added delete image functionality

### PR DESCRIPTION
**Pull request by SergeyGhazaryan**

### CODE REVIEW

File: gta-gallery/src/components/modal.jsx [hunk 1]
```diff
@@ -9,12 +9,20 @@ export const Modal = ({
...
     <DefaultModal visible={isOpen} onCancel={onCancel} onOk={onOk}>
       <div className='title'>{header}</div>
-      {imageValue && <img className='modal-image' src={imageValue} />}
+      <div className='image-container'>
+        {imageValue && <img className='modal-image' src={imageValue} />}
+        {imageValue && (
+          <div className='delete-image' onClick={deleteImage}>
+            x
+          </div>
+        )}
```
Comment (arayik-99) [inline gta-gallery/src/components/modal.jsx]: Put the img and x inside empty fragment to do one checking

**Review by arayik-99 — CHANGES_REQUESTED** [review]

File: gta-gallery/src/components/changePassword/index.jsx [hunk 2]
```diff
@@ -1,8 +1,9 @@
 import React, { useState } from 'react';
 import { useHistory } from 'react-router-dom';
-import { InputFields } from '../inputFields';
 import { Button } from '../button';
 import { changePassword } from '../../services';
+import { Form } from 'antd';
```
Comment (Karen-Avdalyan) [inline gta-gallery/src/components/changePassword/index.jsx]: move it up

File: gta-gallery/src/components/changePassword/index.jsx [hunk 3]
```diff
@@ -54,12 +34,47 @@ export const ChangePassword = () => {
...
+          >
+            <Input type='password' onChange={setNewPasswordValue} />
+          </Form.Item>
+          <Form.Item
+            label='Repeat new password'
+            name='repeatNewPassword'
+            rules={[
+              {
+                required: true,
+                message: 'Please input your repeat new password!',
```
Comment (Karen-Avdalyan) [inline gta-gallery/src/components/changePassword/index.jsx]: wrong english

File: gta-gallery/src/components/editUser/index.jsx [hunk 4]
```diff
@@ -1,57 +1,87 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import { useHistory } from 'react-router-dom';
-import { InputFields } from '../inputFields';
 import { Button } from '../button';
-import { editUser } from '../../services/userService';
+import { editUser, getCurrentUser } from '../../services/userService';
+import { Form, Input } from 'antd';
```
Comment (Karen-Avdalyan) [inline gta-gallery/src/components/editUser/index.jsx]: Move it up. And why you are using here antd Input if you have created your custom one

File: gta-gallery/src/components/editUser/index.jsx [hunk 5]
```diff
@@ -1,57 +1,87 @@
...
+          initialValues={{
+            firstName: firstNameValue,
+            lastName: lastNameValue,
+            username: usernameValue,
+          }}
+        >
+          <Form.Item label='First name' name='firstName'>
+            <Input
+              type='text'
+              onChange={(e) => setFirstNameValue(e.target.value)}
```
Comment (Karen-Avdalyan) [inline gta-gallery/src/components/editUser/index.jsx:60]: in changePassword you have written this in another way

File: gta-gallery/src/components/gtaVersion/index.jsx [hunk 6]
```diff
@@ -1,20 +1,23 @@
 import React from 'react';
-import { NavLink } from 'react-router-dom';
+import { useHistory } from 'react-router';
 import { PopConfirm } from '../popConfirm';
 
 import './styles.scss';
 
 export const GtaVersion = ({ id, index, image, name, onDelete }) => {
+  const history = useHistory();
+  const handleImageClick = () => {
```
Comment (Karen-Avdalyan) [inline gta-gallery/src/components/gtaVersion/index.jsx:10]: add an empty row here

File: gta-gallery/src/components/gtaVersionDetails/index.jsx [hunk 7]
```diff
@@ -3,105 +3,183 @@ import { useParams } from 'react-router-dom';
 import { VersionModal } from '../shared/versionModal';
 import { Button } from '../button';
 import { getVersion, updateVersion } from '../../services/gtaVersionsService';
+import { Form, Input, Upload } from 'antd';
```
Comment (Karen-Avdalyan) [inline gta-gallery/src/components/gtaVersionDetails/index.jsx]: same here

File: gta-gallery/src/components/gtaVersionDetails/index.jsx [hunk 8]
```diff
@@ -3,105 +3,183 @@ import { useParams } from 'react-router-dom';
...
 
 import './styles.scss';
 
+const normFile = (e) => {
```
Comment (Karen-Avdalyan) [inline gta-gallery/src/components/gtaVersionDetails/index.jsx]: change the name

File: gta-gallery/src/components/header/index.jsx [hunk 9]
```diff
@@ -47,16 +46,19 @@ export const Header = () => {
...
+      result += lastName.charAt(0);
+    }
+    return result;
+  }, [firstName, lastName]);
 
   useEffect(() => {
     getUser();
   }, [authenticated]);
-
   const menu = (
```
Comment (Karen-Avdalyan) [inline gta-gallery/src/components/header/index.jsx]: add an empty row here

File: gta-gallery/src/components/login/index.jsx [hunk 10]
```diff
@@ -1,10 +1,12 @@
...
+import { useHistory } from 'react-router-dom';
 import { useDispatch } from 'react-redux';
-import { InputFields } from '../inputFields';
 import { Button } from '../button';
-import { login } from '../../store/auth/actions';
+import { Input } from '../input';
+import { login, setUser } from '../../store/auth/actions';
 import { login as loginRequest } from '../../services';
+import { getCurrentUser } from '../../services/userService';
+import { Form } from 'antd';
```
Comment (Karen-Avdalyan) [inline gta-gallery/src/components/login/index.jsx]: move it up

File: gta-gallery/src/components/signup/index.jsx [hunk 11]
```diff
@@ -1,10 +1,13 @@
 import React, { useState } from 'react';
 import { useHistory } from 'react-router-dom';
 import { useDispatch } from 'react-redux';
-import { InputFields } from '../inputFields';
 import { Button } from '../button';
-import { signup as defaultSignup } from '../../store/auth/actions';
 import { signup } from '../../services';
+import { Form } from 'antd';
```
Comment (Karen-Avdalyan) [inline gta-gallery/src/components/signup/index.jsx]: move it up

File: gta-gallery/src/components/signup/index.jsx [hunk 12]
```diff
@@ -84,12 +56,54 @@ export const Signup = () => {
...
+            name='password'
+            rules={[{ required: true, message: 'Please input your password!' }]}
+          >
+            <Input type='password' onChange={setPasswordValue} />
+          </Form.Item>
+          <Form.Item
+            label='Repeat password'
+            name='repeatPassword'
+            rules={[
+              { required: true, message: 'Please input your repeat password!' },
```
Comment (Karen-Avdalyan) [inline gta-gallery/src/components/signup/index.jsx]: wrong english

File: gta-gallery/src/layout/index.jsx [hunk 13]
```diff
@@ -2,14 +2,25 @@ import React, { useState, useEffect } from 'react';
 import { GtaVersion } from '../components/gtaVersion';
 import { VersionModal } from '../components/shared/versionModal';
 import { Button } from '../components/button';
+import { Input } from '../components/input';
 import { getAllVersions, createVersion, deleteVersion } from '../services';
+import { Form, Upload } from 'antd';
```
Comment (Karen-Avdalyan) [inline gta-gallery/src/layout/index.jsx]: move it up

File: gta-gallery/src/layout/index.jsx [hunk 14]
```diff
@@ -2,14 +2,25 @@ import React, { useState, useEffect } from 'react';
...
 
 import './styles.scss';
 
+const normFile = (e) => {
```
Comment (Karen-Avdalyan) [inline gta-gallery/src/layout/index.jsx]: change the name

**Review by Karen-Avdalyan — CHANGES_REQUESTED** [review]

File: gta-gallery/src/components/gtaVersion/index.jsx (hunk 6 above)
Comment (SergeyGhazaryan) [reply gta-gallery/src/components/gtaVersion/index.jsx:9]: I know.

File: gta-gallery/src/components/editUser/index.jsx (hunk 5 above)
Comment (SergeyGhazaryan) [reply gta-gallery/src/components/editUser/index.jsx:60]: I know.

File: gta-gallery/src/components/editUser/index.jsx [hunk 15]
```diff
@@ -1,57 +1,87 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import { useHistory } from 'react-router-dom';
-import { InputFields } from '../inputFields';
+import { Form, Input } from 'antd';
 import { Button } from '../button';
-import { editUser } from '../../services/userService';
+import { editUser, getCurrentUser } from '../../services/userService';
+import { setUser } from '../../store/auth/actions';
+import { useDispatch } from 'react-redux';
```
Comment (Karen-Avdalyan) [inline gta-gallery/src/components/editUser/index.jsx]: move it up

File: gta-gallery/src/components/gtaVersionDetails/index.jsx (hunk 7 above)
Comment (Karen-Avdalyan) [inline gta-gallery/src/components/gtaVersionDetails/index.jsx]: move it up

File: gta-gallery/src/components/gtaVersionDetails/index.jsx [hunk 16]
```diff
@@ -3,105 +3,183 @@ import { useParams } from 'react-router-dom';
 import { VersionModal } from '../shared/versionModal';
 import { Button } from '../button';
 import { getVersion, updateVersion } from '../../services/gtaVersionsService';
+import { Form, Input, Upload } from 'antd';
 
 import './styles.scss';
 
+const getValueFromEvent = (e) => {
```
Comment (Karen-Avdalyan) [inline gta-gallery/src/components/gtaVersionDetails/index.jsx]: maybe it would be better to rename it to getFileList

File: gta-gallery/src/components/header/index.jsx (hunk 9 above)
Comment (Karen-Avdalyan) [inline gta-gallery/src/components/header/index.jsx]: please

File: gta-gallery/src/components/signup/index.jsx [hunk 17]
```diff
@@ -1,10 +1,12 @@
 import React, { useState } from 'react';
 import { useHistory } from 'react-router-dom';
 import { useDispatch } from 'react-redux';
-import { InputFields } from '../inputFields';
 import { Button } from '../button';
-import { signup as defaultSignup } from '../../store/auth/actions';
 import { signup } from '../../services';
+import { Form, Input } from 'antd';
```
Comment (Karen-Avdalyan) [inline gta-gallery/src/components/signup/index.jsx]: move it up

File: gta-gallery/src/layout/index.jsx [hunk 18]
```diff
@@ -1,15 +1,25 @@
 import React, { useState, useEffect } from 'react';
+import { Form, Upload, Input } from 'antd';
 import { GtaVersion } from '../components/gtaVersion';
 import { VersionModal } from '../components/shared/versionModal';
 import { Button } from '../components/button';
 import { getAllVersions, createVersion, deleteVersion } from '../services';
 
 import './styles.scss';
 
+const getValueFromEvent = (e) => {
```
Comment (Karen-Avdalyan) [inline gta-gallery/src/layout/index.jsx]: maybe it would be better to rename it to getFileList

**Review by Karen-Avdalyan — CHANGES_REQUESTED** [review]

File: gta-gallery/src/components/gtaVersionDetails/index.jsx [hunk 19]
```diff
@@ -1,107 +1,185 @@
...
 import { useParams } from 'react-router-dom';
+import { Form, Input, Upload } from 'antd';
 import { VersionModal } from '../shared/versionModal';
 import { Button } from '../button';
 import { getVersion, updateVersion } from '../../services/gtaVersionsService';
 
 import './styles.scss';
 
+const getFileList = (e) => {
+  if (Array.isArray(e)) {
```
Comment (arayik-99) [inline gta-gallery/src/components/gtaVersionDetails/index.jsx]: You can remove the curly braces

File: gta-gallery/src/components/gtaVersionDetails/index.jsx [hunk 20]
```diff
@@ -1,107 +1,185 @@
...
+          imageUrl,
+          loading: false,
+        })
+      );
+    }
+  };
+
+  const beforeUpload = (file) => {
+    const isJpgOrPng = file.type === 'image/jpeg' || file.type === 'image/png';
+    if (!isJpgOrPng) {
```
Comment (arayik-99) [inline gta-gallery/src/components/gtaVersionDetails/index.jsx]: remove curly braces

File: gta-gallery/src/components/gtaVersionDetails/index.jsx [hunk 21]
```diff
@@ -1,107 +1,185 @@
...
+      console.error('You can only upload JPG/PNG file!');
+    }
+    const isLt2M = file.size / 1024 / 1024 < 2;
+    if (!isLt2M) {
```
Comment (arayik-99) [inline gta-gallery/src/components/gtaVersionDetails/index.jsx]: remove curly braces

File: gta-gallery/src/components/header/index.jsx [hunk 22]
```diff
@@ -47,16 +46,20 @@ export const Header = () => {
...
-      0
-    )}`;
-  }, [currentUser]);
+    let result = '';
+    if (firstName) {
+      result += firstName.charAt(0) + ' ';
+    }
+    if (lastName) {
+      result += lastName.charAt(0);
+    }
```
Comment (arayik-99) [inline gta-gallery/src/components/header/index.jsx]: Do we have both firstName and lastName as mandatory fields? If so we can just check if either of them don't exist and just return.

File: gta-gallery/src/layout/index.jsx [hunk 23]
```diff
@@ -1,15 +1,25 @@
...
+import { Form, Upload, Input } from 'antd';
 import { GtaVersion } from '../components/gtaVersion';
 import { VersionModal } from '../components/shared/versionModal';
 import { Button } from '../components/button';
 import { getAllVersions, createVersion, deleteVersion } from '../services';
 
 import './styles.scss';
 
+const getFileList = (e) => {
+  if (Array.isArray(e)) {
```
Comment (arayik-99) [inline gta-gallery/src/layout/index.jsx]: Remove curly braces

File: gta-gallery/src/layout/index.jsx [hunk 24]
```diff
@@ -92,6 +70,46 @@ const Layout = () => {
...
+  const getBase64 = (img, callback) => {
+    const reader = new FileReader();
+    reader.addEventListener('load', () => callback(reader.result));
+    reader.readAsDataURL(img);
+  };
+
+  const handleChange = (info) => {
+    if (info.file.status === 'uploading') {
+      setImageValue({ loading: true });
+      return;
```
Comment (arayik-99) [inline gta-gallery/src/layout/index.jsx]: you can shorten this to return setImageValue...

**Review by arayik-99 — CHANGES_REQUESTED** [review]

File: gta-gallery/src/components/gtaVersionDetails/index.jsx [hunk 25]
```diff
@@ -1,107 +1,179 @@
...
+import { Form, Input, Upload } from 'antd';
 import { VersionModal } from '../shared/versionModal';
 import { Button } from '../button';
 import { getVersion, updateVersion } from '../../services/gtaVersionsService';
 
 import './styles.scss';
 
+const getFileList = (e) => {
+  if (Array.isArray(e)) return e;
+  return e && e.fileList;
```
Comment (Shitikyan) [inline gta-gallery/src/components/gtaVersionDetails/index.jsx]: what this line means?

File: gta-gallery/src/components/gtaVersionDetails/index.jsx [hunk 26]
```diff
@@ -1,107 +1,179 @@
...
-  const [gtaVersionDetails, setGTAVersionDetails] = useState({});
+  const [form] = Form.useForm();
   const [updateModalVisible, setUpdateModalVisible] = useState(false);
-  const [imageValue, setImageValue] = useState('');
+  const [imageValue, setImageValue] = useState({
+    loading: false,
+  });
   const [nameValue, setNameValue] = useState('');
   const [informationValue, setInformationValue] = useState('');
+  const [version, setVersion] = useState({
```
Comment (Shitikyan) [inline gta-gallery/src/components/gtaVersionDetails/index.jsx:25]: Can you remove this version state?

File: gta-gallery/src/layout/index.jsx [hunk 27]
```diff
@@ -92,6 +68,42 @@ const Layout = () => {
...
+      getBase64(file.originFileObj, (imageUrl) =>
+        setImageValue({
+          imageUrl,
+          loading: false,
+        })
+      );
+    }
+  };
+
+  const beforeUpload = (file) => {
```
Comment (Shitikyan) [inline gta-gallery/src/layout/index.jsx]: This method we have in 2 places. Please move to other file and use the same method all places

File: gta-gallery/src/layout/index.jsx [hunk 28]
```diff
@@ -92,6 +68,42 @@ const Layout = () => {
     setVersions(versionsArray);
   };
 
+  const getBase64 = (img, callback) => {
```
Comment (Shitikyan) [inline gta-gallery/src/layout/index.jsx]: This method we have in 2 places. Please move to other file and use the same method all places

File: gta-gallery/src/components/gtaVersionDetails/index.jsx (hunk 26 above)
Comment (SergeyGhazaryan) [reply gta-gallery/src/components/gtaVersionDetails/index.jsx:25]: No, I show it.

File: gta-gallery/src/components/gtaVersionDetails/index.jsx (hunk 25 above)
Comment (SergeyGhazaryan) [reply gta-gallery/src/components/gtaVersionDetails/index.jsx]: I changed that.
